src/shared/ui/Input.tsx

```diff
@@ -1,24 +1,16 @@
-import styled from "@emotion/styled";
-
-export type InputProps = React.ComponentProps<"input">;
-
-export const Input = styled.input<InputProps>`
-    width: ${({ width }) => width};
-    height: ${({ height }) => height};
-
-    padding: 8px 0px;
-
-    border: none;
-    border-bottom: 1px solid;
-    border-color: ${({ theme }) => theme.colors.gray.gray400};
-
-    font-size: 1rem;
-
-    outline: none;
-
-    transition: border-color 100ms ease-in-out;
-
-    &:focus {
-        border-color: ${({ theme }) => theme.colors.gray.gray900};
-    }
-`;
+import { forwardRef, Fragment } from "react";
+
+import * as Styles from "@/shared/ui/Input.styled";
+
+export interface InputProps extends React.ComponentProps<"input"> {
+    error?: string;
+}
+
+export const Input = forwardRef<HTMLInputElement, InputProps>(({ error, ...props }, ref) => {
+    return (
+        <Fragment>
+            <Styles.InputElement type="text" ref={ref} error={error} {...props} />
+            <Styles.InputErrorMessage>{error}</Styles.InputErrorMessage>
+        </Fragment>
```

Comment:
Fragment를 import하지 않아도 아래와 같이 표현하면 불필요한 import를 줄일 수 있어서 보통 아래와 같이 사용하곤 하는데요. 별도로 Fragment를 import해서 사용하신 이유가 있으시면 말씀 주세요! 

```suggestion
        <>
            <Styles.InputElement type="text" ref={ref} error={error} {...props} />
            <Styles.InputErrorMessage>{error}</Styles.InputErrorMessage>
        </>
```

Reply:
사실 별다른 이유는 없고... <> 이게 안이뻐서 Fragment 로 넣었습니다
찾아보니 배열 순회시 key 값을 지정하려면 Fragment 를 사용해야되는군요..!!